flink-table/flink-table-api-java-bridge/src/main/java/org/apache/flink/table/api/java/internal/StreamTableEnvironmentImpl.java

```diff
@@ -104,7 +106,9 @@ public static StreamTableEnvironment create(
 			settings.getBuiltInCatalogName(),
 			new GenericInMemoryCatalog(settings.getBuiltInCatalogName(), settings.getBuiltInDatabaseName()));
 
-		FunctionCatalog functionCatalog = new FunctionCatalog(catalogManager);
+		ModuleManager moduleManager = new ModuleManager();
+
+		FunctionCatalog functionCatalog = new FunctionCatalog(catalogManager, moduleManager);
```

Comment:
I saw repeated pattern like this. It's probably better to have a constructor that doesn't take a module manager and instead creates it inside the constructor.

Reply:
TableEnv needs to get a hold on module manager to be able to load/unload module

Reply:
Got it.